psydac/linalg/basic.py

```diff
@@ -52,6 +52,20 @@ def dot(self, a, b):
 
         """
 
+    @abstractmethod
+    def axpy(self, a, x, y):
+        """
+        This method compute the operation x+=a*y.
```

Comment:
This is not correct, it should be `y += a * x`. See the BLAS documentation [here](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/axpy.html).

Reply:
Ah I thought the name was strange for the purpose of the function

Reply:
This PR correct it, Fortunately, we just have to change the order of the arguement in `mul_iadd` method to make all the previous call work again:
[4e98e4c](https://github.com/pyccel/psydac/pull/310/commits/4e98e4c52fd97dac04a948466a0dc7180cdc4239)

Reply:
This PR correct a typo I didn't see 
 [6133446](https://github.com/pyccel/psydac/pull/310/commits/613344625a0e5b965475f651a3491f68ec5896ef)